apis/cluster/serverpool.go

```diff
@@ -34,6 +34,7 @@ type ServerPool struct {
 	Name              string      `json:"name,omitempty"`
 	Image             string      `json:"image,omitempty"`
 	Size              string      `json:"size,omitempty"`
+	SpotPrice         string      `json:"spotPrice,omitempty"`
```

Comment:
So a change here is a change to the API.. Because we plan on refactoring the API anyway I think it's okay to start introducing changes like this now. But in the future it won't be nearly as easy.

My only reservation is that we might be introducing an AWS level API field into an API that is aimed at being provider agnostic. Maybe preface this field name with `Aws` such as `AwsSpotPrice` so users will know that this is only for AWS?

Regardless of if other clouds support a pricing change like this, I think calling it `AwsSpotPrice` for now works as this is the only implementation that uses it.

However this does beg the question of whether or not we want to have cloud objects in the long run..

Reply:
On second thought we can just keep this as is, and have a validation check for passing it in with other clouds.

Do you mind adding a quick check to our validation?

Reply:
Currently it's AWS only - but same can work with Google's `preemptible` support (will get there as well). 

I am currently away for a few days and not able to work on this - as far as I remember if you don't pass anything it's ignored. Can you/someone please add the validation meanwhile - before the master diverges too much :)

Reply:
Let's merge and open a ticket! Enjoy your holiday!

Reply:
Thank you - @ganlaksh is volunteering to add the validation meanwhile for #347